### ISSUE TITLE
Multi-Trip Flag per Vehicle

### ISSUE THREAD

**Issue by JoeHeming** [VROOM-Project/vroom#492]
Hello @jcoupey , I wanted to know if it's possible to allow only some vehicles to participate in multiple trips. I already checked previous issues and as far as I understand, the shipments object could be used per task to define its pickup details along with its delivery details. That being said, once all shipments are defined and all vehicles have full capacity, I want to allow only certain vehicles to go back to depot according to something like a flag "true" for allowing multiple trips if it can be within their time window. Is there any possible solution for this currently or perhaps a work around? Thanks in advance.

**Comment by jcoupey** [comment]
No, if modelling reload/deliveries using shipments, I don't see a way to prevent a given vehicle to go back to the depot. You may use other constraints to try and avoid it, such as reducing time window (not enough time for a second trip), reducing the scope of what the vehicle can handle (using skills) or try to limit the number of tasks in the route (that would be #421 not implemented yet).

But it would probably require quite some tuning depending on your use-case, and provide no guarantee.

For the record, can you provide some more context on what kind of situation requires such a constraint?

**Comment by JoeHeming** [comment]
Having an outsourced fleet, where each vehicle has different contract terms; some can do multiple trips but others can't even if they have a time window. So if there are tasks (each is unique in terms of component items) that exceed the capacity of vehicles, then the vehicles should go on a trip first to deliver tasks then get back to depot to reload and go on another trip if the time window of the vehicles allows that. In that case I want to consider only a certain number of vehicles (allowed to do so) not all of them. Also, I don't want to limit the number of tasks per vehicle nor reduce the time window in order to have the best utilization of the capacity. 

If I wanted to take this forward can I use the unassigned tasks after running the algorithm the first time to run again for these tasks using the vehicles allowed to do so and deducting from the start of their time window the duration of the first trip +  service time for loading?

If you have a better suggestion to solve this issue, I'd really appreciate it. Thank you.

**Comment by JoeHeming** [comment]
Hello @jcoupey sorry to bother you, but do you have any suggestion regarding my issue? You help and support are highly appreciated. Thank you.

**Comment by jcoupey** [comment]
> can I use the unassigned tasks after running the algorithm the first time to run again for these tasks using the vehicles allowed to do so

Yes you can probably come up with something iterative one way or another. But it's really hard to tell whether this will turn out to be efficient globally. I suspect the answer to that is also very dependent on your use-case.

**Comment by JoeHeming** [comment]
My use case is that I have a list of orders and the fleet as described above (some vehicles can do more than one trip and others can't). I want to deliver orders so I need to do shipments to accommodate for pickup and delivery.

I found also weird behavior that I want to control, where vehicles go back to depot even if there are still some picked up orders still not delivered. I want to make sure they don't go back to depot unless they're empty and if they do go, they leave with maximum capacity that could fit. How can this be solved?

I noticed vehicle step object, could this be used in an abstract manner to describe how I want the vehicle to behave. For example, start -> pickup -> deliver -> pickup -> deliver -> end. This way I am setting the "multi-trip" option for the vehicle to 2 trips. Is this a possible option?

**Comment by jcoupey** [comment]
> My use case is that I have a list of orders and the fleet as described above

When I mean use-case, I mean that the whole problem description is at stake: presence/absence/tightness of various constraints such as capacities, amounts, time windows etc. and their conjunction. It's simply not possible to provide a specific support in a GH issue, especially without knowing all of this. I already agreed that trying an iterative approach might work, but the only way to know is to go on and try. I'd be happy to discuss the outcome if you do some testing and can share the results.

> I found also weird behavior that I want to control, where vehicles go back to depot even if there are still some picked up orders still not delivered.

Please don't use a ticket for follow-up questions on other topics because it makes the whole discussion hard to follow. Feel free to open a dedicated ticket on this with some data that allows reproducing the behavior that is puzzling you.

> I noticed vehicle step object

As stated [in the docs](https://github.com/VROOM-Project/vroom/blob/master/docs/API.md#vehicles) the `steps` object for vehicles only makes sense in plan mode, that is when you provide the expected route order yourself in input. I suspect this is not what you need as you seem to want VROOM to decide the tasks ordering for you.
